packages/terra-switch/src/Switch.jsx

```diff
@@ -0,0 +1,145 @@
+import React, { useState, useRef } from 'react';
+import PropTypes from 'prop-types';
+import uniqueid from 'lodash.uniqueid';
+import classNames from 'classnames/bind';
+import * as KeyCode from 'keycode-js';
+import styles from './Switch.module.scss';
+
+const cx = classNames.bind(styles);
+
+const SWITCH_STATE = Object.freeze({
+  ON: 'On',
+  OFF: 'Off',
+});
+
+const propTypes = {
+  /**
+    * Whether or not the Switch is enabled ("ON").
+    */
+  isOn: PropTypes.bool,
+
+  /**
+    * If true, the user won't be able to toggle the switch.
+    */
+  isDisabled: PropTypes.bool,
+
+  /**
+    * Text of the label.
+    */
+  labelText: PropTypes.node.isRequired,
+
+  /**
+    * Callback fired when the state is changed. Parameters are (event).
+    */
+  onChange: PropTypes.func,
+};
+
+const defaultProps = {
+  isOn: false,
+  isDisabled: false,
+  onChange: undefined,
+};
+
+const Switch = ({
+  isOn,
+  isDisabled,
+  onChange,
+  labelText,
+  ...customProps
+}) => {
+  const [focused, setFocus] = useState(false);
+  const sliderButtonRef = useRef(null);
+
+  const handleClick = (event) => {
+    setFocus(true);
+    sliderButtonRef.current.focus();
+    onChange(event);
+  };
+
+  const handleOnKeyDown = (event) => {
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      event.preventDefault();
+      setFocus(true);
```

Comment:
shouldn't we call the onchange here ?

Reply:
No, because calling `onChange()` here will cause switch to toggle multiple times on long press hence `onChange()` is being called in` handleOnKeyUp()`